evm/src/cpu/kernel/interpreter.rs

```diff
@@ -396,9 +448,29 @@ impl<'a> Interpreter<'a> {
         self.running = true;
         while self.running {
             let pc = self.generation_state.registers.program_counter;
-            if self.is_kernel() && self.halt_offsets.contains(&pc) {
+
+            if let Some(halt_context) = self.halt_context {
+                if self.is_kernel()
+                    && self.halt_offsets.contains(&pc)
+                    && halt_context == self.generation_state.registers.context
+                {
+                    log::debug!(
+                        "se acabo el show cuando el contepsto era {:?}",
```

Comment:
I think you have a spelling mistake here: `contesto` yeah?

Reply:
Yeah let's keep it to the universal language here 😆

Reply:
ops!